cypress/e2e/fee-fine/overdue-fine-renewed-triggers.cy.js

```diff
@@ -0,0 +1,427 @@
+    returnedAfterRecurring: createNoticeTemplate(
+      `Autotest_${getRandomPostfix()}_Overdue_fine_renewed_after_recurring`
+    ),
+  };
+  const selectOptions = (template) => {
+    const generalOptions = {
+      noticeName: 'FeeFine',
+      noticeId: 'feeFine',
+      format: 'Email',
+      action: 'Overdue fine, renewed',
+      templateName: template.name,
+    };
+    if (template.name === noticeTemplates.returnedUponAt.name) {
+      return {
+        send: 'Upon/At',
+        ...generalOptions,
+      };
+    } else if (template.name === noticeTemplates.returnedAfterOnce.name) {
+      return {
+        send: 'After',
+        sendBy: {
+          duration: '1',
+          interval: 'Minute(s)',
+        },
+        frequency: 'One Time',
+        ...generalOptions,
+      };
+    } else if (template.name === noticeTemplates.returnedAfterRecurring.name) {
+      return {
+        send: 'After',
+        sendBy: {
+          duration: '1',
+          interval: 'Minute(s)',
+        },
+        frequency: 'Recurring',
+        sendEvery: {
+          duration: '1',
+          interval: 'Minute(s)',
+        },
+        ...generalOptions,
+      };
+    }
+    return generalOptions;
+  };
+  const searchResultsData = (templateName) => {
+    return {
+      userBarcode: userData.barcode,
+      itemBarcode: itemData.barcode,
+      object: 'Notice',
+      circAction: 'Send',
+      // TODO: add check for date with format <C6/8/2022, 6:46 AM>
+      servicePoint: testData.userServicePoint.name,
+      source: 'System',
+      desc: `Template: ${templateName}. Triggering event: Overdue fine renewed.`,
+    };
+  };
+  const checkNoticeIsSent = (checkParams) => {
+    SearchPane.searchByUserBarcode(userData.barcode);
+    SearchPane.findResultRowIndexByContent(checkParams.desc).then((rowIndex) => {
+      SearchPane.checkResultSearch(checkParams, rowIndex);
+    });
+  };
+  const createPatronNoticeTemplate = (template) => {
+    NewNoticePolicyTemplate.startAdding();
+    NewNoticePolicyTemplate.checkInitialState();
+    NewNoticePolicyTemplate.addToken('item.title');
+    NewNoticePolicyTemplate.create(template, false);
+    NewNoticePolicyTemplate.chooseCategory(template.category);
+    NewNoticePolicyTemplate.checkPreview();
+    NewNoticePolicyTemplate.saveAndClose();
+    NewNoticePolicyTemplate.waitLoading();
+    template.category = 'AutomatedFeeFineCharge';
+    NewNoticePolicyTemplate.checkAfterSaving(template);
+  };
+  const duplicatePatronNoticeTemplate = (template) => {
+    NewNoticePolicyTemplate.duplicateTemplate();
+    NewNoticePolicyTemplate.typeTemplateName(template.name);
+    NewNoticePolicyTemplate.typeTemplateSubject(template.subject);
+    NewNoticePolicyTemplate.checkPreview();
+    NewNoticePolicyTemplate.saveAndClose();
+    NewNoticePolicyTemplate.waitLoading();
+    template.category = 'AutomatedFeeFineCharge';
+    NewNoticePolicyTemplate.checkAfterSaving(template);
+  };
```

Comment:
Let's move all these methods to the appropriate fragment

Reply:
done